osm2gtfs/core/configuration.py

```diff
@@ -31,6 +33,64 @@ def __init__(self, args):
         # Validate and prepare start and end date
         self._prepare_dates()
 
+        # Initiate variable for schedule source information
+        self._schedule_source_file = None
+
+    def get_schedule_source(self, refresh=False):
+        """Loads the schedule source information.
+
+        Loads a schedule source file from either a path or a url specified
+        in the config file
+
+        :return schedule_source_file: The schedule source file object.
+
+        """
+
+        if 'schedule_source' not in self.config:
+            sys.stderr.write("Error: No schedule source specified in config file.\n")
+            sys.exit(0)
```

Comment:
Like @nlehuby I prefer this to return `None` instead of bailing out.

Reply:
I'm still strongly in favour of explicitly opting out. This goes down to the very basic vision of the script, e.g. look at osm2gtfs' subtitle *Turn OpenStreetMap data **and schedule information** into GTFS*. 

To have an implementation that accepts by default configuration files without any schedule source for me is not a good practice, kind of against the idea of GTFS and encourages to produce low quality GTFS. I do think it should be possible to do so, but then you have to explicitly state this in the configuration file.

Reply:
But weren't advocating for putting as much information into OSM? For frequency based regions, I don't see why they shouldn't define the frequency of a line in OSM and don't have to bother to specify this source.

If you allow `None` here, you might as well allow it to not be there.

Reply:
Yes, as much information that should be in OSM. And especially to stick to the standards by the community. And full schedule information is clearly not desired to be in OpenStreetMap. For frequency there is no accepted nor proposed tag (standard), that I'm aware of. In other words, this is not expected and mostly not wanted to be added to OSM (unfortunately). Exactly because of this rationale I'm advocating to provide a standard in osm2gtfs to get it in at the moment when it is the right place to feed it in.

Maybe we can agree on a compromise:

* We make `schedule_source` optional.
* The standard `schedule_creator` (not the configuration class) throws an error, if there is no file specified.

This would basically follow your opinion, but (at least for the standard creator) do the same checks, as it does currently.

Reply:
Sounds like a good compromise to me. @nlehuby?